utils/gyb.py

```diff
@@ -304,7 +309,7 @@ def splitGybLines(sourceLines):
     dedents = 0
     try:
         for tokenKind, tokenText, tokenStart, (tokenEndLine, tokenEndCol), lineText \
-            in tokenize.generate_tokens(sourceLines.__iter__().next):
+            in tokenize.generate_tokens(partial(next, sourceLines.__iter__())):
```

Comment:
That would be fine except the result of `next(sourceLinesIter)` is not callable. Taken from [`tokenize.generate_tokens` docs](https://docs.python.org/2/library/tokenize.html#tokenize.generate_tokens)

>  ... requires one argument, readline, which must be a callable object which provides the same interface as the readline() method of built-in file objects (see section File Objects) ...

 Notice that in the original code `next` was sent to `tokenize.generate_tokens` not `next()`, i.e., something that is callable. @frewsxcv suggestion is the equivalent of sending `next()` and not `next`. Hence why I went with the partial function application.

### Testing

I first applied this patch:

```patch
diff --git a/utils/gyb.py b/utils/gyb.py
index ab422f1..68a8280 100755
--- a/utils/gyb.py
+++ b/utils/gyb.py
@@ -307,9 +307,10 @@ def splitGybLines(sourceLines):
     unmatchedIndents = []

     dedents = 0
+    sourceLinesIter = iter(sourceLines)
     try:
         for tokenKind, tokenText, tokenStart, (tokenEndLine, tokenEndCol), lineText \
-            in tokenize.generate_tokens(partial(next, sourceLines.__iter__())):
+            in tokenize.generate_tokens(next(sourceLinesIter)):

             if tokenKind in (tokenize.COMMENT, tokenize.ENDMARKER):
                 continue
```

Then I ran this from the `utils` directory.

```python
$ python3
>>> from gyb import *
>>> src = splitLines('''\
... if x:
...     print x
... if y: # trailing comment
...     print z
...     if z: # another comment\
... ''')
>>> splitGybLines(src)
Traceback (most recent call last):
File "<stdin>", line 1, in <module>
File "/Users/lovelett/Source/swift/utils/gyb.py", line 313, in splitGybLines
  in tokenize.generate_tokens(next(sourceLinesIter)):
File "/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/tokenize.py", line 514, in _tokenize
  line = readline()
TypeError: 'str' object is not callable
```

Expected:

```python
$ python3
>>> from gyb import *
>>> src = splitLines('''\
... if x:
...     print x
... if y: # trailing comment
...     print z
...     if z: # another comment\
... ''')
>>> splitGybLines(src)
[3, 5]
```